keras_retinanet/bin/train.py

```diff
@@ -151,74 +149,47 @@ def create_generators(args):
     # create random transform generator for augmenting training data
     transform_generator = random_transform_generator(flip_x_chance=0.5)
 
-    if args.dataset_type == 'coco':
-        # import here to prevent unnecessary dependency on cocoapi
-        from ..preprocessing.coco import CocoGenerator
+    generators = None
 
-        train_generator = CocoGenerator(
-            args.coco_path,
-            'train2017',
-            transform_generator=transform_generator,
-            batch_size=args.batch_size
-        )
-
-        validation_generator = CocoGenerator(
-            args.coco_path,
-            'val2017',
-            batch_size=args.batch_size
-        )
-    elif args.dataset_type == 'pascal':
-        train_generator = PascalVocGenerator(
-            args.pascal_path,
-            'trainval',
-            transform_generator=transform_generator,
-            batch_size=args.batch_size
-        )
-
-        validation_generator = PascalVocGenerator(
-            args.pascal_path,
-            'test',
-            batch_size=args.batch_size
-        )
-    elif args.dataset_type == 'csv':
-        train_generator = CSVGenerator(
-            args.annotations,
-            args.classes,
-            transform_generator=transform_generator,
-            batch_size=args.batch_size
-        )
+    for plugin in PluginManagerSingleton.get().getAllPlugins():
+        try:
+            getattr(plugin.plugin_object, 'dataset_type')
+        except AttributeError:
+            print("Plugin ({}) does not contain a definition for dataset_type and cannot be utilised.".format(plugin.name))
```

Comment:
I think it would make sense to use the warning [1] module for this. If we ever expose some way to modify the warning handling, that can be used to let the user turn these into errors on request.

[1] https://docs.python.org/3/library/warnings.html

Reply:
I looked at using `logging.warning` for things like this. But I was considering some built-in logger/warning system from Python, but saw no evidence of its use currently, so held off. Again, it can always be future PRs which introduce logging on a wider scale than just this feature. Definitely agree on the use of logging/warning though over debug prints.

Reply:
It's done internally in a few places, but not in the scripts now. Still, never to early to start doing things the right way.

But it can be a separate PR if you prefer (and it doesn't have to be you doing it ^^).

Reply:
Volunteering? ;)

Reply:
Wasn't really my intention, no :)